src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -166,10 +167,18 @@ private static HttpConnectionKey GetConnectionKey(HttpRequestMessage request, Ur
                 Debug.Assert(HttpUtilities.IsSupportedNonSecureScheme(proxyUri.Scheme));
                 if (sslHostName == null)
                 {
-                    // Standard HTTP proxy usage for non-secure requests
-                    // The destination host and port are ignored here, since these connections
-                    // will be shared across any requests that use the proxy.
-                    return new HttpConnectionKey(HttpConnectionKind.Proxy, null, 0, null, proxyUri);
+                    if (HttpUtilities.IsNonSecureWebSocketScheme(uri.Scheme))
```

Comment:
Note: In this code block, we force tunneling if there is a proxy. (the `SslProxyTunnel` state, and the new introduced `ProxyTunnel` state for ws)